### PR TITLE
Consolidate serializer tests

### PR DESCRIPTION
**Pull request by shariarriday**

### CODE REVIEW

File: libcaf_core/caf/serialization.test.cpp [hunk 1]
```diff
@@ -0,0 +1,767 @@
...
+  }
+
+private:
+  std::vector<std::string> parse_value(const std::string& list_val,
+                                       const std::string& delimiter) {
+    const std::regex delimiter_regex{delimiter};
+    return std::vector<std::string>{
+      std::sregex_token_iterator(list_val.cbegin(), list_val.cend(),
+                                 delimiter_regex, -1),
+      {}};
```
Comment (Neverlord) [inline libcaf_core/caf/serialization.test.cpp:633]: I think using a `regex` here is overkill. Isn't this simply a `split`? https://github.com/actor-framework/actor-framework/blob/master/libcaf_core/caf/string_algorithms.hpp#L35

**Review by Neverlord — CHANGES_REQUESTED** [review]
Just one nitpick.

File: libcaf_core/caf/serialization.test.cpp (hunk 1 above)
Comment (shariarriday) [reply libcaf_core/caf/serialization.test.cpp:633]: Thanks, that is helpful. I was not aware of the existing split functions. I am working on updating the PR accordingly.

**Review by Neverlord — APPROVED** [review]
Thanks for wrapping this up! 👍

**Pull request merged**